src/merkle_tree.rs

```diff
@@ -22,100 +24,116 @@ impl MerkleTree {
     pub fn new<T: std::convert::AsRef<[u8]>>(data: &[T]) -> MerkleTree {
         let initial_blocks: Vec<HashedData> = data
             .iter()
-            .map(|value| {
-                Sha3_256::digest(value).into()
-            })
+            .map(|value| Sha3_256::digest(value).into())
```

Comment:
Maybe `.map(Sha3_256::digest).map(HashedData::from)` of the top of my head. Altho this is a bit more nitpicky

Reply:
I believe I'll refactor that section all together once I finish the project. (If everything goes right, that'll be sooner rather than later).

I'll certainly keep it in mind when the time comes.